platform/darwin/src/MGLStyleLayerManager.h

```diff
@@ -0,0 +1,45 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLStyleLayer.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+namespace mbgl {
+    namespace style {
+        class Layer;
+        class LayerFactory;
+    }
+}
+
+@protocol MGLStyleLayerFactory
+
+/**
+Initializes and returns a layer with a raw pointer to the backing store,
+associated with a style.
+*/
+- (MGLStyleLayer *)createWithRawLayer:(mbgl::style::Layer *)rawLayer;
```

Comment:
In the existing layers, `createWithRawLayer` doesn't appear to touch any instance properties. Is that intended in the future? 

If not, perhaps consider making this a class method.

Reply:
The more "Cocoa" name would be something like `styleLayerWithRawLayer:`, but Swift favors using `make` for factory methods. 

So perhaps we consider something like?

`+ (MGLStyleLayer *)makeStyleLayerWithRawLayer:(mbgl::style::Layer *)rawLayer;`

Reply:
And if we do go with a class method, that raises the question of whether we need the ObjC factory class in the first place. The `MGL<X>StyleLayer` could have a static variable for the rawLayerFactory...

However, I tend to favor a separate factory class - as I believe it may evolve to manage its child layers.

Reply:
I also question the need for a factory class if it ends up being stateless – standalone C functions can accomplish the same thing without the cognitive overhead of an Objective-C object. If a factory class pattern is strictly necessary for some reason, we could also implement it as a C++ class.